protocol/test/forge/v2/strategies/RamosStrategy.t.sol

```diff
@@ -0,0 +1,458 @@
+            assertEq(assetBalances.length, 3);
+            assertEq(assetBalances[0].asset, reportedAssets[0]);
+            assertApproxEqAbs(assetBalances[0].balance, (initialDaiBalance+50) * initialBptTotalSupply / (initialBptTotalSupply+100) - 50, 1);
+            assertEq(assetBalances[1].asset, reportedAssets[1]);
+            assertEq(assetBalances[1].balance, 2*100+50);
+            assertEq(assetBalances[2].asset, reportedAssets[2]);
+            assertEq(assetBalances[2].balance, 2*200-50);
+            assertEq(debt, 100e18);
+        }
+    }
+}
+
+contract RamosStrategyTestBorrowAndRepay is RamosStrategyTestBase {
+    uint256 public constant trvStartingBalance = 10e18;
+    uint256 public constant borrowCeiling = 1.01e18;
+
+    event BorrowAndAddLiquidity(uint256 amount);
+    event RemoveLiquidityAndRepay(uint256 amount);
+
+    function setUp() public {
+        _setUp();
+
+        // Add the new strategy, and setup TRV such that it has stables to lend and issue dUSD.
+        // Add the `executor` as an operator of RAMOS strategy
+        vm.startPrank(executor);
+        trv.addNewStrategy(address(strategy), borrowCeiling, 0);
+        strategy.setAssets(reportedAssets);
+        deal(address(dai), address(trv), trvStartingBalance, true);
+        dUSD.addMinter(address(trv));
+
+        strategy.setOperator(executor);
+        vm.stopPrank();
+
+        // Add the strategy as an operator of RAMOS
+        address ramosOwner = RAMOS(RAMOS_ADDRESS).owner();
+        startHoax(ramosOwner);
+        RAMOS(RAMOS_ADDRESS).setOperator(address(strategy));
+        vm.stopPrank();
+    }
+
+    function getRamosBalances() internal view returns (uint256 bptTotalSupply, uint256 bptBalance, uint256 templeBalance, uint256 stableBalance) {
+        bptTotalSupply = IBPT(address(bptToken)).getActualSupply();
+
+        if (bptTotalSupply > 0) {
+            bptBalance = amoStaking.stakedBalance() + bptToken.balanceOf(address(amoStaking));
+
+            (uint256 templeBalanceInLP, uint256 stableBalanceInLP) = PoolHelper(POOL_HELPER_ADDRESS).getTempleStableBalances();
+            templeBalance = templeBalanceInLP * bptBalance / bptTotalSupply;
+            stableBalance = stableBalanceInLP * bptBalance / bptTotalSupply;
+        }
+    }
+
+    function test_borrowAndAddLiquidity() public {
+        uint256 amount = 1e18;
+        uint256 slippageBps = 100;  // 1%
+        ITempleStrategy.AssetBalance[] memory assetBalances;
+
+        assertEq(dai.balanceOf(address(trv)), trvStartingBalance);
+
+        (uint256 debt, uint256 available, uint256 ceiling) = strategy.trvBorrowPosition();
+        assertEq(debt, 0);
+        assertEq(available, borrowCeiling);
+        assertEq(ceiling, borrowCeiling);
+
+        (, uint256 bptBalanceBefore, uint256 templeBalanceBefore, uint256 stableBalanceBefore) = getRamosBalances();
+        (assetBalances,) = strategy.latestAssetBalances();
+        uint256 daiBalanceBefore = assetBalances[0].balance;
+
```

Comment:
I think it would be worth adding another `querySwap()` for a large amount of liquidity proportional to the pool size (ie 10 million worth) -- and checking that the ratio of assets before is the same as the ratio after the join (and another for exit)

It's important that we've nailed this logic or we might affect the pool in a bad way which would cost $$